File: src/AdventureWorks.Application/DataEngine/HumanResources/JobCandidate/Commands/Queries/GetJobCandidates/JobCandidateLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.HumanResources;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.HumanResources.JobCandidate.Queries.GetJobCandidates
{
    public partial class JobCandidateLookupModel : IHaveCustomMapping
    {
        public int JobCandidateID { get; set; }
        public int? EmployeeID { get; set; }
        public string Resume { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.HumanResources.JobCandidate, JobCandidateLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<JobCandidateLookupModel, BaseJobCandidate>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/human-resources/job-candidate/job-candidate-lookup-model.ts

export interface IJobCandidateLookupModel {
  jobCandidateID: number;
  employeeID: number | null;
  resume: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class JobCandidateLookupModel implements IJobCandidateLookupModel {
  jobCandidateID: number;
  employeeID: number | null;
  resume: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<JobCandidateLookupModel>) {
    if (!!init) {
      Object.assign<JobCandidateLookupModel, Partial<JobCandidateLookupModel>>(this, init);
    }
  }
}
